internal/core/operations/remove_rooms/remove_rooms_executor.go

```diff
@@ -24,24 +24,51 @@ package remove_rooms
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/topfreegames/maestro/internal/core/entities/operation"
 	"github.com/topfreegames/maestro/internal/core/operations"
+	"github.com/topfreegames/maestro/internal/core/services/room_manager"
+	"go.uber.org/zap"
 )
 
-type RemoveRoomsExecutor struct{}
+type RemoveRoomsExecutor struct {
+	roomManager *room_manager.RoomManager
+}
 
-func NewExecutor() *RemoveRoomsExecutor {
-	return &RemoveRoomsExecutor{}
+func NewExecutor(roomManager *room_manager.RoomManager) *RemoveRoomsExecutor {
+	return &RemoveRoomsExecutor{roomManager}
 }
 
-func (e *RemoveRoomsExecutor) Execute(_ context.Context, _ *operation.Operation, _ operations.Definition) error {
-	// TODO(gabrielcorado): implement it
+func (e *RemoveRoomsExecutor) Execute(ctx context.Context, op *operation.Operation, definition operations.Definition) error {
+	removeDefiniton := definition.(*RemoveRoomsDefinition)
+	rooms, err := e.roomManager.ListRoomsWithDeletionPriority(ctx, op.SchedulerName, removeDefiniton.Amount)
+	if err != nil {
+		return fmt.Errorf("failed to list rooms to delete: %w", err)
+	}
+
+	logger := zap.L().With(
+		zap.String("scheduler_name", op.SchedulerName),
```

Comment:
@gabrielcorado what do you think about including the `service` key in this logger as we are doing in other places? Ex: zap.String("service", "worker")

Reply:
Does it make sense in this file to add this reference? The operation itself doesn't know who is actually calling it. If we assume where it is going to run, I'd be in favor of leaving without it and, since we are deploying our workers in different commands, we can add those labels into the global instance (which we derive from);

Reply:
I was thinking more about how it will be when making queries for logs, say that we want to see logs filtering the worker cmd, without this label would It be possible? I saw that we were populating this `service` label in some logger parents, I'm worried that those generated logs would "escape" from the standard we are creating with those labels. Adding this label in the global instance seems to be the better approach here.